### PR TITLE
Adding test for api

### PR DESCRIPTION
**Pull request by jeffin143**
Here are samples of test

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+              done();
+            });
+      });
+
+      it('it should fail to POST a article without title', (done) => {
+        let article = {
+          description :  'writing test is always good',
+          author : 'justin',
+          tags :  'hello'
+        }
```
Comment (jeffin143) [inline test/test.js:67]: @sastaachar , I am not sure how we are taking the request, object 

Is it a object or
```
[title]=
[author]=
```

I guess
this format is good
```
{
author:
title:
}
```

Also take a look why this test is failing , 
it should throw an error

Comment (jeffin143) [reply test/test.js:67]: I will take a look, if you didn't beat me to it :)

Comment (dinolinjob) [reply test/test.js:67]: test fails at 1) it should fail to POST a article without title

Comment (sastaachar) [reply test/test.js:67]: title is missing there so an internal error will thrown i.e 400 ,
but there was a typo so it should work now

**Comment by jeffin143** [comment]
Server is not connected to mongo db ,
Do you have .env file and mongo db server started ?

**Comment by jeffin143** [comment]
It passes in my local and also in ci checks

**Comment by dinolinjob** [comment]
> Server is not connected to mongo db ,
> Do you have .env file and mongo db server started ?

Sorry, my bad. My mongodb service was stopped..
It works cool

**Comment by jeffin143** [comment]
@dinolinjob , If everything looks of to you, could you approved this PR

**Comment by dinolinjob** [comment]
@jeffin143 using mocha for testing is much better than using jest I feel. So, I don't think there's a necessity for going with #7 . What do you suggest?

**Comment by jeffin143** [comment]
> @jeffin143 using mocha for testing is much better than using jest I feel. So, I don't think there's a necessity for going with #7 . What do you suggest?

Yes jest, isn't needed for API Testing, we would follow mocha for API test , jest would be used for unit testing or functional testing after the code refactoring

**Comment by dinolinjob** [comment]
> > @jeffin143 using mocha for testing is much better than using jest I feel. So, I don't think there's a necessity for going with #7 . What do you suggest?
> 
> Yes jest, isn't needed for API Testing, we would follow mocha for API test , jest would be used for unit testing or functional testing after the code refactoring

Yes, Sure

**Comment by jeffin143** [comment]
> shouldn't it be `if(process.env.NODE_ENV !== 'test')`

Make that change in next pr

**Pull request merged**